test/e2e/addon/certmanager.go

```diff
@@ -0,0 +1,218 @@
+package addon
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	certmanagerv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	certmanagerclientset "github.com/cert-manager/cert-manager/pkg/client/clientset/versioned"
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientgo "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	ik8s "github.com/aws/eks-hybrid/internal/kubernetes"
+	"github.com/aws/eks-hybrid/test/e2e/kubernetes"
+)
+
+const (
+	certManagerNamespace      = "cert-manager"
+	certManagerName           = "cert-manager"
+	certManagerCainjectorName = "cert-manager-cainjector"
+	certManagerWebhookName    = "cert-manager-webhook"
+	certName                  = "test-cert"
+	certTestNamespace         = "cert-test"
+	issuerName                = "selfsigned-issuer"
+	certSecretName            = "selfsigned-cert-tls"
+	certManagerWaitTimeout    = 5 * time.Minute
+)
+
+// CertManagerTest tests the cert-manager addon
+type CertManagerTest struct {
+	Cluster    string
+	addon      *Addon
+	K8S        clientgo.Interface
+	EKSClient  *eks.Client
+	K8SConfig  *rest.Config
+	Logger     logr.Logger
+	CertClient certmanagerclientset.Interface
+}
+
+// Create installs the cert-manager addon
+func (c *CertManagerTest) Create(ctx context.Context) error {
+	c.addon = &Addon{
+		Cluster:   c.Cluster,
+		Namespace: certManagerNamespace,
+		Name:      certManagerName,
+	}
+
+	if err := c.addon.CreateAndWaitForActive(ctx, c.EKSClient, c.K8S, c.Logger); err != nil {
+		return err
+	}
+
+	// Wait for cert-manager deployment to be ready
+	if err := kubernetes.DeploymentWaitForReplicas(ctx, certManagerWaitTimeout, c.K8S, c.addon.Namespace, certManagerName); err != nil {
+		return err
+	}
+
+	// Wait for cert-manager webhook deployment to be ready
+	if err := kubernetes.DeploymentWaitForReplicas(ctx, certManagerWaitTimeout, c.K8S, c.addon.Namespace, certManagerWebhookName); err != nil {
+		return err
+	}
+
+	// Wait for cert-manager cainjector deployment to be ready
+	if err := kubernetes.DeploymentWaitForReplicas(ctx, certManagerWaitTimeout, c.K8S, c.addon.Namespace, certManagerCainjectorName); err != nil {
+		return err
+	}
+
+	c.Logger.Info("Cert-manager setup is complete")
+	return nil
+}
+
+// Validate tests cert-manager functionality by creating and validating a self-signed certificate
+func (c *CertManagerTest) Validate(ctx context.Context) error {
+	c.Logger.Info("Starting cert-manager validation")
+
+	// Create test namespace if it doesn't exist
+	if err := kubernetes.CreateNamespace(ctx, c.K8S, certTestNamespace); err != nil {
+		return fmt.Errorf("failed to create test namespace: %v", err)
```

Comment:
```suggestion
		return fmt.Errorf("failed to create test namespace: %w", err)
```

Use %w to format `err` so that it can be unwrapped later.